lib/NullsToUndefined.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * A type which ensures that all `null` values become `undefined`.
+ *
+ * See: https://stackoverflow.com/questions/50374869/generic-way-to-convert-all-instances-of-null-to-undefined-in-typescript
+ */
+export type NullsAsUndefined<T> = T extends null
```

Comment:
There may be cases where we want to explicitly mark something as "null". Can't think of any atm though, just something to look out for in the future

Reply:
Maybe, but let's consider that most other languages don't have a divide between null and undefined, but rather just a unified null value. Would prefer to lean towards keeping everything as undefined if possible since undefined has a nicer syntax than dealing with null.